server/src/core/server/services/users/helpers.ts

```diff
@@ -104,3 +104,65 @@ export function checkForNewUserEmailDomainModeration(
   }
   return null;
 }
+
+interface EmailIsAliasResult {
+  base?: {
+    value: string;
+    start: string;
+    domain: string;
+  } | null;
+  alias?: {
+    value: string;
+    start: string;
+    domain: string;
+  } | null;
+  isAlias: boolean;
+}
+
+export function emailIsAlias(email?: string | null): EmailIsAliasResult {
+  if (!email) {
+    return {
+      base: null,
+      alias: null,
+      isAlias: false,
+    };
+  }
+
+  const split = email.split("@");
+  if (split.length < 2) {
+    return {
+      base: null,
+      alias: null,
+      isAlias: false,
+    };
+  }
+
+  const domain = split[2];
+  const aliasStart = split[0];
+
+  const aliasSplit = email.split("+");
+  if (aliasSplit.length <= 1) {
+    return {
+      base: null,
+      alias: null,
+      isAlias: false,
+    };
+  }
+
+  const baseOfAlias = aliasSplit[0];
+  const baseEmail = `${baseOfAlias}@${domain}`;
+
+  return {
+    base: {
+      value: baseEmail,
```

Comment:
I am trying to think if there's a way to make these namings any clearer. I had to take some time to think through what each was still. Maybe something like the following could help clarify?:
`baseEmail` --> `baseOfAliasWithDomain`
`aliasStart` --> `aliasWithoutDomain`

Reply:
fixed